### PR TITLE
fix linting and go versions

### PR DESCRIPTION
**Pull request by byott**
Snyk local output:

✔ Tested 27 dependencies for known issues, no vulnerable paths found.

### CODE REVIEW

File: interface.go [hunk 1]
```diff
@@ -8,11 +8,11 @@ import (
...
 // MessageProcessor processes a consumed message. Implementors are responsible for unmarshalling the data
 type MessageProcessor interface {
-	ProcessMessage(ctx context.Context, record *kinesis.Record) MessageProcessorError
+	ProcessMessage(ctx context.Context, record *kinesis.Record) Error
 }
 
-// MessageProcessorError represents an error that can be used to indicate to the consumer that an error should be retried
-type MessageProcessorError interface {
+// Error represents an error that can be used to indicate to the consumer that an error should be retried
+type Error interface {
```
Comment (willyjfarrell) [inline interface.go]: Question: Why did this need to change?

Comment (byott) [reply interface.go]: make lint was giving error:

`type name will be used as messageprocessor.MessageProcessorError by other packages, and that stutters; consider calling this Error (golint)
`

**Review by willyjfarrell — DISMISSED** [review]

File: interface.go (hunk 1 above)
Comment (gcase555) [reply interface.go]: We might consider a more descriptive name, like `MessageError` (or something else) 

does that work? I think there is some concern this `Error` could be confused with another golang Error interface

Comment (byott) [reply interface.go]: Sounds good - just updated 👍

**Review by willyjfarrell — APPROVED** [review]

**Review by gcase555 — APPROVED** [review]

**Pull request merged**